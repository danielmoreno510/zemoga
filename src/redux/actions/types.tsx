// Rulings --------------------------------------------------------------------------
export enum RULINGS {
    FETCH_RULING_LIST = 'FETCH_RULING_LIST',
    PUT_RULING_LIST = 'PUT_RULING_LIST',
    PUT_OPTION_RULING_LIST = 'PUT_OPTION_RULING_LIST',
    FETCH_RULING_DETAILS = 'FETCH_RULING_DETAILS',
    PUT_RULING_DETAILS = 'PUT_RULING_DETAILS',
    FETCH_RULING_OPERATION = 'FETCH_RULING_OPERATION',
    PUT_RULING_OPERATION= 'PUT_RULING_OPERATION'
}