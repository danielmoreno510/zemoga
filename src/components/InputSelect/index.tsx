import InputSearch from './InputSelect'

export default InputSearch