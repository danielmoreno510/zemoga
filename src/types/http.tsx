interface IHttpError {
  error: {
    message: string;
    name: string;
    stack: string;
    config: {
      transitional: {
        silentJSONParsing: true;
        forcedJSONParsing: true;
        clarifyTimeoutError: false;
      };
      adapter: string[];
      transformRequest: any[];
      transformResponse: any[];
      timeout: number;
      xsrfCookieName: string;
      xsrfHeaderName: string;
      maxContentLength: number;
      maxBodyLength: number;
      env: any;
      headers: {
        Accept: string;
      };
      baseURL: string;
      withCredentials: boolean;
      method: string;
      url: string;
    };
    code: string;
    status: number;
  };
}
