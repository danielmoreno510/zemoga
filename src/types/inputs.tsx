export enum OPTIONS {
    GRID = 'Grid',
    LIST = 'List',
  }

  export enum OPERATION {
    ADD = 'ADD',
    SUBS = 'SUBS',
    NULL = ''
  }