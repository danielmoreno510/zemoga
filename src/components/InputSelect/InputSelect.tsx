import { OPTIONS } from "@/types/inputs";
import { useState } from "react";

const InputSelect: React.FC<{ value: OPTIONS; setValue: (value: OPTIONS) => void }> = ({ value, setValue }) => {
  const [open, setOpen] = useState<boolean>(false);

  const changeValue = (option: OPTIONS) => {
    setOpen(false)
    setValue(option)
  }

  return (
    <div>
      <button
        data-testid="open"
        id="dropdownDelayButton"
        data-dropdown-toggle="dropdownDelay"
        data-dropdown-delay="500"
        data-dropdown-trigger="hover"
        className="text-black border-2 border-black border-solid outline-none font-medium text-sm px-5 py-2.5 text-center inline-flex items-center"
        type="button"
        onClick={() => setOpen(!open)}
      >
        {value}{" "}
        <svg
          className="w-2.5 h-2.5 ml-2.5"
          aria-hidden="true"
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 10 6"
        >
          <path
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            d="m1 1 4 4 4-4"
          />
        </svg>
      </button>
      {open && (
        <div
          id="dropdownDelay"
          className="z-10 bg-white divide-y divide-gray-100 border-2 border-black border-solid border-t-white shadow"
        >
          <ul
            className="py-2 text-sm text-gray-700"
            aria-labelledby="dropdownDelayButton"
          >
            <li
              data-testid="list"
              className="block px-4 py-2 text-black cursor-pointer"
              onClick={() => changeValue(OPTIONS.LIST)}
            >
              List
            </li>
            <li
              data-testid="grid"
              className="block px-4 py-2 text-black cursor-pointer"
              onClick={() => changeValue(OPTIONS.GRID)}
            >
              Grid
            </li>
          </ul>
        </div>
      )}
    </div>
  );
};

export default InputSelect;
