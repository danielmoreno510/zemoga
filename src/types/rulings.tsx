import { RULINGS } from "@/redux/actions/types";
import { CallEffect, PutEffect } from "redux-saga/effects";
import { OPERATION, OPTIONS } from "./inputs";

export interface IRulingListResponse {
  data: IRuling[];
}

export interface IRuling {
  id: string;
  name: string;
  description: string;
  category: string;
  picture: string;
  lastUpdated: string;
  votes: {
    positive: number;
    negative: number;
  };
  percent?: {
    positive: number;
    negative: number;
  };
}

export type TRulingEffect =
  | PutEffect<{ type: string }>
  | CallEffect<IRulingListResponse>;

export interface IRulingInitialState {
  isFetchingRulingList?: boolean;
  rulingList?: IRuling[];
  option?: OPTIONS;
  isFetchingRulingDetails?: boolean;
  rulingDetails?: IRuling | null;
}

export interface IRulingReducer {
  type: RULINGS;
  rulingList?: IRuling[];
  option?: OPTIONS;
  id?: string;
  rulingDetails?: IRuling | null;
  operation?: string;
  ruling?: IRuling;
}

export interface IRulingListStateProps {
  rulings: IRulingInitialState;
}

export interface IRulingListDispatchProps {
  getRulings?: () => any;
  changeOption?: (option: OPTIONS) => void;
  getRulingDetails?: (id: string) => void;
  fetchVote?: (operation: OPERATION, ruling: IRuling) => void;
}

export interface IRulingCard {
  ruling: IRuling;
  view: OPTIONS;
  saveVote: (operation: OPERATION) => void;
}
