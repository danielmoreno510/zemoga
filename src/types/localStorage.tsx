import { IRuling } from "./rulings";

export enum KeysEnum {
  RulingList = 'RulingList',
  RulingDetails = 'RulingDetails',
}

export interface IStorageRulingDetails {
  expiry: number;
  data: IRuling;
}
