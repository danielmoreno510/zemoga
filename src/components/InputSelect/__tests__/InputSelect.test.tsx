import { fireEvent, render } from "@testing-library/react";
import InputSelect from "../InputSelect";
import { OPTIONS } from "@/types/inputs";
import React from "react";

describe("InputSelect component", () => {
  describe("GIVEN a set of valid props", () => {
    const testProps = {
      value: OPTIONS.GRID,
      setValue: jest.fn(),
    };

    describe("WHEN the component is rendered", () => {
      const container = render(<InputSelect {...testProps} />);
      const open = container.getAllByTestId("open");
      fireEvent.click(open[0])

      it("THEN should display a regular Home", () => {
        expect(container.container).toMatchSnapshot();
      });
    });

    describe("WHEN the list button is pressed", () => {
      React.useState = jest.fn().mockReturnValue([true, jest.fn()])
      const container = render(<InputSelect {...testProps} />);
      const list = container.getAllByTestId("list");
      fireEvent.click(list[0])

      it('THEN setValue function should be called', () => {
        expect(testProps.setValue).toHaveBeenCalledWith(OPTIONS.LIST);
      });
    });

    describe("WHEN the grid button is pressed", () => {
      React.useState = jest.fn().mockReturnValue([true, jest.fn()])
      const container = render(<InputSelect {...testProps} />);
      const grid = container.getAllByTestId("grid");
      fireEvent.click(grid[0])

      it('THEN setValue function should be called', () => {
        expect(testProps.setValue).toHaveBeenCalledWith(OPTIONS.GRID);
      });
    });
    
  });
});
