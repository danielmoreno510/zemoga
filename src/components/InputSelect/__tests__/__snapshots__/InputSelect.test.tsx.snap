// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelect component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Home 1`] = `
<div>
  <div>
    <button
      class="text-black border-2 border-black border-solid outline-none font-medium text-sm px-5 py-2.5 text-center inline-flex items-center"
      data-dropdown-delay="500"
      data-dropdown-toggle="dropdownDelay"
      data-dropdown-trigger="hover"
      data-testid="open"
      id="dropdownDelayButton"
      type="button"
    >
      Grid
       
      <svg
        aria-hidden="true"
        class="w-2.5 h-2.5 ml-2.5"
        fill="none"
        viewBox="0 0 10 6"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1 1 4 4 4-4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="z-10 bg-white divide-y divide-gray-100 border-2 border-black border-solid border-t-white shadow"
      id="dropdownDelay"
    >
      <ul
        aria-labelledby="dropdownDelayButton"
        class="py-2 text-sm text-gray-700"
      >
        <li
          class="block px-4 py-2 text-black cursor-pointer"
          data-testid="list"
        >
          List
        </li>
        <li
          class="block px-4 py-2 text-black cursor-pointer"
          data-testid="grid"
        >
          Grid
        </li>
      </ul>
    </div>
  </div>
</div>
`;
